hcloud/error.go

```diff
@@ -100,6 +101,15 @@ type Error struct {
 }
 
 func (e Error) Error() string {
+	if resp := e.Response(); resp != nil {
+		isServerError := resp.StatusCode >= http.StatusInternalServerError && resp.StatusCode <= 599
+		hasCorrelationID := resp.InternalCorrelationID() != ""
+		if isServerError && hasCorrelationID {
+			// For easier debugging, the error string contains the Correlation ID if the response
+			// was sent with a status code in the "server error" range (500-599).
+			return fmt.Sprintf("%s (%s) (Correlation ID: %s)", e.Message, e.Code, resp.InternalCorrelationID())
+		}
```

Comment:
Why not always print the correlation ID ?

Reply:
If the error is probably caused by the user (4xx) there is no need to contact support and for us to check it out. Only if the server (5xx) is responsible we need to investigate further.

Of course, there are often issues where an invalid error code is sent.

Reply:
Changed it to any code.